src/device_info.sh

```diff
@@ -0,0 +1,186 @@
+#
+# @target Target machine
+function get_disk()
+{
+  local target="$1"
+  local cmd
+  local info
+  local size
+  local mount
+  local fs
+
+  cmd="df -h / | tail -n 1 | tr -s ' '"
+  case "$target" in
+    2) # LOCAL_TARGET
+      info=$(cmd_manager 'SILENT' "$cmd")
+      ;;
+  esac
+
+  fs="$(echo "$info" | cut -d' ' -f1)"
+  size="$(echo "$info" | cut -d' ' -f2)"
+  mount="$(echo "$info" | cut -d' ' -f6)"
+
+  shared_data['disk_size']="$size"
+  shared_data['fs_name']="$fs"
+  shared_data['fs_mount']="$mount"
+}
+
+# This function populates the os variable from the shared_data variable.
+#
+# @target Target machine
+function get_os()
+{
+  local target="$1"
+  local os
+
+  case "$target" in
+    2) # LOCAL_TARGET
+      os=$(detect_distro '/')
+      ;;
+  esac
+
+  shared_data['os']="$os"
+}
+
+# This function calls other functions to populate the shared_data variable with
+# the data related to the hardware from the target machine.
+#
+# @target Target machine
+function process_all_data()
+{
+  local target="$1"
+
+  get_ram "$target"
+  get_cpu "$target"
+  get_disk "$target"
+  get_os "$target"
+}
+
+# This function shows the information stored in the shared_data variable.
+function display_data()
+{
+  say 'CPU:'
+  printf '  Model: %s\n' "${shared_data['cpu_model']}"
+  if [[ -n "${shared_data['cpu_currently']}" ]]; then
+    printf '  Current frequency (MHz): %s\n' "${shared_data['cpu_currently']}"
+  fi
+  if [[ -n "${shared_data['cpu_max']}" ]]; then
+    printf '  Max frequency (MHz): %s\n' "${shared_data['cpu_max']}"
+  fi
+  if [[ -n "${shared_data['cpu_min']}" ]]; then
+    printf '  Min frequency (MHz): %s\n' "${shared_data['cpu_min']}"
+  fi
+  say 'RAM:'
+  printf '  Total RAM: %s\n' "$(numfmt --from=si --to=iec "${shared_data['ram']}K")"
+  say 'Storage devices:'
+  printf '  Filesystem: %s\n' "${shared_data['fs_name']}"
+  printf '  Size: %s\n' "${shared_data['disk_size']}"
+  printf '  Mounted on: %s\n' "${shared_data['fs_mount']}"
+  say 'Operating System:'
+  printf '  Distribution: %s\n' "${shared_data['os']}"
+}
+
+# This function parses the options provided to 'kw device', makes the necessary
+# ajustments, and then call other functions to process and display the hardware
+# information. If no argument is provided, then the function assigns the value
+# from configurations[default_deploy_target] to the option variable; if there is
+# no value there either, then option is by default assigned to local.
+function device_info()
+{
+  local option="$1"
+  local target
+
+  if [[ -z "$option" && -n "${configurations[default_deploy_target]}" ]]; then
+    option='--'"${configurations[default_deploy_target]}"
+  fi
+  option=${option:-'--local'}
+
+  case "$option" in
+    --local)
+      target="$LOCAL_TARGET"
+      ;;
+    -h)
+      device_info_help
+      exit 0
+      ;;
+    *)
+      complain "Invalid option: $option"
+      return 22 # EINVAL
+      ;;
+  esac
+
+  process_all_data "$target"
+  display_data
```

Comment:
I would prefer that we separate the parser from the function that manages the action. In other words, create a new function dedicated to parser command-line options (e.g., `device_info_parser`) and keep this function as an entry point for `device_info`. You probably want to put this as the first function in this file.